components/toast/script.js

```diff
@@ -37,82 +55,146 @@ class GamefaceToast extends BaseComponent {
-
-        if (!containersCreated) this.createToastContainers();
-
         components.loadResource(this)
             .then(this.init)
             .catch(err => console.error(err));
     }
 
+    disconnectedCallback() {
+        this.detachListeners();
+        this.isRendered = false;
+    }
+
     attachEventListeners() {
-        if (this._targetElement ) {
-            this._targetElement.addEventListener('click', this.show);
+        this.addEventListener('animationend', this.handleAnimationEnd);
+    }
+
+    detachListeners() {
+        this.removeEventListener('animationend', this.handleAnimationEnd);
+    }
+    /* eslint-enable require-jsdoc */
+
+    /**
+     * Custom element lifecycle method. Called when an attribute is changed.
+     * @param {string} name
+     * @param {string} oldValue
+     * @param {string|boolean|array} newValue
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (!this.isRendered) return;
+
+        this.updateAttributeState(name, newValue);
+    }
+
+    /**
+     * Will update the state properties linked with the checkbox attributes
+     * @param {string} name
+     * @param {string|boolean|array} value
+     */
+    updateAttributeState(name, value) {
+        switch (name) {
+            case 'position':
+            case 'gravity':
+            case 'timeout':
+                this.updateState(name, value);
+                break;
+            case 'target':
+                this.updateTargetState(name, value);
+                break;
         }
+    }
 
-        if (!animationEventAttached) {
-            document.addEventListener('animationend', (event) => {
-                if (event.animationName === 'guic-toast-fade-out') event.target.parentElement.removeChild(event.target);
-            });
-            animationEventAttached = true;
+    /**
+     * Update the toasts's state.
+     * @param {string} name - the name of the prop
+     * @param {string | boolean} value - the value of the the prop
+     * @returns {void}
+     */
+    updateState(name, value) {
+        if (!this.isStatePropValid(name, value)) return;
+        this.state[name] = value;
+    }
+
+    /**
+     * Update the toasts's state.
+     * @param {string} name - the name of the prop
+     * @param {string | boolean} value - the value of the the prop
+     * @returns {void}
+     */
+    updateTargetState(name, value) {
+        if (!this.isStatePropValid(name, value)) return;
+
+        const newTarget = document.querySelector(value);
+
+        if (this.state.target && newTarget) {
+            this.state.target.removeEventListener('click', this.show);
+        }
+
+        if (newTarget) {
+            this.state.target = newTarget;
+            this.state.target.addEventListener('click', this.show);
+        } else {
+            console.error(`Can't find an element with the selector "${value}".`);
         }
     }
```

Comment:
```suggestion
        const newTarget = document.querySelector(value);
        if (!newTarget) return console.error(`Can't find an element with the selector "${value}".`);

        if (this.state.target) this.state.target.removeEventListener('click', this.show);

        this.state.target = newTarget;
        this.state.target.addEventListener('click', this.show);
    }
```

Its more clean in this way